web/Services/AdminService.cs

```diff
@@ -98,22 +98,25 @@ private async Task<List<UserDbModel>> GetPendingAdmins()
         return pendingAdmins;
     }
 
-    public async Task<List<LandlordDbModel>> GetLandlordDisplayList(string approvalStatus)
+    public async Task<List<LandlordDbModel>> GetLandlordList(LandlordListModel landlordListModel)
     {
-        return approvalStatus switch
+        var landlordQuery = landlordListModel.Approved switch
         {
-            "Unapproved" => await _dbContext.Landlords.Where(u => u.CharterApproved == false).ToListAsync(),
-            "Approved" => await _dbContext.Landlords.Where(u => u.CharterApproved == true).ToListAsync(),
-            _ => await _dbContext.Landlords.ToListAsync(),
+            true => _dbContext.Landlords.Where(l => l.CharterApproved == true),
+            false => _dbContext.Landlords.Where(l => l.CharterApproved == false),
+            _ => _dbContext.Landlords
         };
+        
+        landlordQuery = landlordListModel.Assigned switch
+        {
+            true => landlordQuery.Where(l => _dbContext.Users.SingleOrDefault(u => u.LandlordId == l.Id) != null),
+            false => landlordQuery.Where(l => _dbContext.Users.SingleOrDefault(u => u.LandlordId == l.Id) == null),

```

Comment:
Would you mind changing these to `_dbContext.Users..Any` and `!_dbContext.Users..Any` so they don't break if we add multiple users for each landlord?

Reply:
This section of code got slightly rewritten, and when I try changing SingleOrDefault to Any it causes issues with non-equivalent variable types - there might be a way around this, but I'm not certain it's urgent. Maybe we could figure this out in the future, if we get around to adding multiple users per landlord? It's just that I know Tom is waiting for this branch to merge in order to rebase some styling work onto it, so I don't want to delay that for too long.

Reply:
It's now been fixed - see my comment below. I think I misunderstood how Any worked (a quick google didn't help), so was using it the wrong way and thus causing the error.